tests/Microsoft.Identity.Test.Unit/PublicApiTests/ExtensiblityTests.cs

```diff
@@ -218,6 +218,157 @@ public async Task ValidateAppTokenProviderAsync()
+                parameters.TryGetValue("additional_param2", out additionalParam2);
+                parameters.TryGetValue("additional_param3", out additionalParam3);
+                parameters.TryGetValue("additional_param4", out additionalParam4);
+                parameters.TryGetValue("additional_param5", out additionalParam5);
+
+                Assert.AreEqual("value1", additionalParam1);
+                Assert.AreEqual("value2", additionalParam2);
+                Assert.AreEqual("value3", additionalParam3);
+#if SUPPORTS_SYSTEM_TEXT_JSON
+                Assert.AreEqual("[\"GUID\", \"GUID2\", \"GUID3\"]", additionalParam4);
+#else
+                Assert.AreEqual("[\"GUID\",\"GUID2\",\"GUID3\"]", additionalParam4);
+#endif
+                Assert.AreEqual("{\"value5json\":\"value5\"}", additionalParam5);
+
+                Assert.AreEqual((IReadOnlyDictionary<string, string>)parameters, result.AdditionalResponseParameters);
+
+                //Ensure not all cache parameters are required
+                app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                        .WithClientSecret(TestConstants.ClientSecret)
+                        .WithExperimentalFeatures(true)
+                        .WithAuthority("https://login.microsoftonline.com/tid/")
+                        .WithHttpManager(httpManager)
+                        .BuildConcrete();
+
+                httpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseWithAdditionalParamsMessage();
+
+                result = await app.AcquireTokenForClient(TestConstants.s_scope.ToArray())
+                    .WithAdditionalCacheParameters(new List<string> { "additional_param1", "additional_param3" })
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                parameters = app.AppTokenCacheInternal.Accessor.GetAllAccessTokens().Single().PersistedCacheParameters;
+                Assert.IsTrue(parameters.Count == 2);
+
+                parameters.TryGetValue("additional_param1", out additionalParam1);
+                parameters.TryGetValue("additional_param3", out additionalParam3);
+
+                Assert.AreEqual("value1", additionalParam1);
+                Assert.AreEqual("value3", additionalParam3);
+
+                Assert.AreEqual((IReadOnlyDictionary<string, string>)parameters, result.AdditionalResponseParameters);
+
+                //Ensure missing cache parameters are not added
+                app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                      .WithClientSecret(TestConstants.ClientSecret)
+                      .WithExperimentalFeatures(true)
+                      .WithAuthority("https://login.microsoftonline.com/tid/")
+                      .WithHttpManager(httpManager)
+                      .BuildConcrete();
+
+                httpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseWithAdditionalParamsMessage();
```

Comment:
do we need a test for expired cache handling?

Reply:
Added a test case for it